internal/registry/registry.go

```diff
@@ -432,35 +442,56 @@ func (r *Registry) SetPolicy(policyCfg config.Policy) error {
 }
 
 // AllowPeer configures the policy to allow messages published by the
-// identified peer. Returns true if policy changed.
+// identified peer. Returns true if policy changed. This configuration is not
+// persisted across indexer restarts. Update the indexer config file to persist
+// the change.
+//
+// If using an assigner service, the allowed peer is persisted in the
+// datastore, as this is the basis of assingment of a publisher to an indexer.
 func (r *Registry) AllowPeer(peerID peer.ID) bool {
-	filePath := "" // TODO: get this from daemon
 	if r.policy.Allow(peerID) {
-		cfg, err := config.Load(filePath)
-		if err != nil {
-			log.Errorw("Cannot load config file", "err", err)
-			return true
-		}
-		cfg.Discovery.Policy = r.policy.ToConfig()
-		err = cfg.Save(filePath)
-		if err != nil {
-			log.Errorw("Cannot save config file", "err", err)
+		if r.useAssigner {
+			err := r.saveAssignedPeer(peerID)
+			if err != nil {
+				log.Errorw("Cannot save assignment", "err", err)
+			}
 		}
 		return true
 	}
 	return false
 }
 
 // BlockPeer configures the policy to block messages published by the
-// identified peer, and block the peer from registering as a provider.
+// identified peer.  Returns true if policy changed. This configuration is not
+// persisted across indexer restarts. Update the indexer config file to persist
+// the change.
+//
+// If using an assigner service, the blocked peer is removed from the persisted
+// set of allowed peers in datastore. This allows unassigning the peer to the
+// indexer.
 func (r *Registry) BlockPeer(peerID peer.ID) bool {
-	return r.policy.Block(peerID)
+	if r.policy.Block(peerID) {
+		if r.useAssigner {
+			err := r.deleteAssignedPeer(peerID)
+			if err != nil {
+				log.Errorw("Cannot delete assignment", "err", err)
+			}
+		}
+		return true
+	}
+	return false
 }
 
-// AllowList returns list of explicitly allowed peer IDs, or false if policy
-// does not have allow list because it allows by default.
-func (r *Registry) AllowList() ([]peer.ID, bool) {
-	return r.policy.AllowList()
+// ListAssignedPeers returns list of allowed peer IDs, when the indexer is
+// configured to work with an assigner service. Otherwise returns error.
+func (r *Registry) ListAssignedPeers() ([]peer.ID, error) {
+	if r.useAssigner {
+		peers, ok := r.policy.ListAllowedPeers()
```

Comment:
Thinking out loud:
What is the rationale for introducing the term "ListAssignedPeers" were it is basically identical to "ListAllowedPeers"?

Looking at the admin API holistically, it seems a bit odd that we have allow, block, list allowed and a separate list assigned.

For simplicity, i wonder if we should stick to either one term or the other. 

If the rationale is to make it clear that "hey this node is being managed by assigner" then a simple admin API that returns the value of `useAssigner` say via `/admin/status` would maybe offer a more robust approach where an assigner (and indeed any process with access to admin API) can check if a node is being managed by an assigner or not and program behaviour accordingly.

If any of this makes sense please feel free to leave TODOs for future PRs. 
What's here seems perfectly functional.

Reply:
It is a rename of `ListAllowedPeers` which was only created for this purpose in a previous PR. This rename makes it clear that it is only used when working with an assigner service.

When not working with an assigner service, this is useless, because the config holds this information. So, we will only have a single `ListAssignedPeers` with corresponding API endpoint.